src/prommis/solvent_extraction/solvent_extraction.py

```diff
@@ -131,24 +127,78 @@ def build(self):
         self.organic_outlet = Port(extends=self.mscontactor.organic_outlet)
 
         def mass_transfer_term(b, t, s, k, l, m):
-            aqueous = b.mscontactor.aqueous
-            phase_list = aqueous.phase_list
-            if s == b.mscontactor.elements.first():
-                return (
-                    b.mscontactor.material_transfer_term[t, s, (k, l, m)]
-                    == -b.mscontactor.aqueous_inlet_state[t].get_material_flow_terms(
-                        phase_list, m
-                    )
-                    * b.partition_coefficient[(k, l, m)]
-                )
+            if self.config.aqueous_to_organic == True:
```

Comment:
You can make this far more compact, and use most of the previous logic to do this. The only thing that changes is which stream you use for the distribution coefficient, so that is all that should be in the `if/else`.

I.e., revert this to how it was and start with:

```
if self.config.aqueous_to_organic:
    state = b.mscontactor.aqueous
    in_state = b.mscontactor.aqueous_inlet_state
else:
    state = b.mscontactor.organic
    in_state = b.mscontactor.organic_inlet_state

phase_list = state.phase_list
```

You can then use `state` and `in_state` in the constraint, and do not need to duplicate all the code.